src/levels/sparsetrianglelevels.jl

```diff
@@ -31,17 +31,7 @@ pattern!(lvl::SparseTriangleLevel{N, Ti}) where {N, Ti} =
 @inline level_default(::Type{<:SparseTriangleLevel{N, Ti, Lvl}}) where {N, Ti, Lvl} = level_default(Lvl)
 data_rep_level(::Type{<:SparseTriangleLevel{N, Ti, Lvl}}) where {N, Ti, Lvl} = DenseData(data_rep_level(Lvl))
 
-function sparsetrianglesize(shape, n)
-    size = 0
-    for dim in 1:n
-        levelsize = 1 
-        for i in 1:dim
-            levelsize *= shape + dim - i - 1
-        end
-        size += fld(levelsize, factorial(dim))
-    end
-    return size + 1
-end
+simplex(shape, n) = fld(prod(shape .+ n .- (1:n)), factorial(n))
```

Comment:
I've reduced all instances of this math to a simplex and virtual simplex function.